src/EventListener/Notification/NotificationsEventListener.php

```diff
@@ -30,55 +36,67 @@ public function __construct(
     public static function getSubscribedEvents(): array
     {
         return [
-            SubmitAddedEvent::class => 'onSubmitAdded',
+            NewSubmitEvent::class => 'onNewSubmit',
             SubmitStatusChangedEvent::class => 'onSubmitStatusChanged',
             ParticipationStatusChangedEvent::class => 'onParticipationStatusChanged',
             NewFeaturedConferenceEvent::class => 'onNewFeaturedConference',
         ];
     }
 
-    public function onSubmitAdded(SubmitAddedEvent $event)
+    public function onNewSubmit(NewSubmitEvent $event)
     {
+        $currentUser = $this->security->getUser();
+
         foreach ($event->getSubmit()->getUsers() as $targetUser) {
-            if ($targetUser === $this->security->getUser()) {
+            if ($targetUser === $currentUser) {
                 continue;
             }
 
-            $notification = new SubmitAddedNotification($event->getSubmit(), $targetUser);
-            $notification->setEmitter($targetUser);
-            $notification->setTrigger(Notification::TRIGGER_SUBMIT_ADDED);
+            new NewSubmitNotification(
+                submit: $event->getSubmit(),
+                emitter: $currentUser,
+                targetUser: $targetUser,
+                trigger: AbstractNotification::TRIGGER_NEW_SUBMIT
+            );
```

Comment:
What do you do with this var?

I have a mixed feeling about this. You relly on the the cascade: persist on the user.notification to make it work.
I would force a persist here to be explicit.

And IIRC, you move all "flush" before dispatching in controllers, so this is never flush?

Reply:
Oh you're right, I originally didn't want to dispatch an event if the flush somehow didn't work, that's why I moved it here but this is true that this is now never flushed.

Is it bad to rely on the cascade if there is one ?

Reply:
It's really personal, but I don't like it. From time to time, bad thing happens.

And anyway:

> explicit is better than implicit

rule 2 of "zen of python" https://www.python.org/dev/peps/pep-0020/ (**MUST READ** / must learn / must apply !)